cms/static/js/views/move_xblock_list.js

```diff
@@ -0,0 +1,213 @@
+],
+function($, Backbone, _, gettext, HtmlUtils, StringUtils, XBlockUtils, MoveXBlockListViewTemplate) {
+    'use strict';
+
+    var XBlockListView = Backbone.View.extend({
+        el: '.xblock-list-container',
+
+        // parent info of currently displayed children
+        parentInfo: {},
+        // currently displayed children XBlocks info
+        childrenInfo: {},
+        // list of visited parent XBlocks, needed for backward navigation
+        visitedAncestors: null,
+
+        // parent to child relation map
+        categoryRelationMap: {
+            course: 'section',
+            section: 'subsection',
+            subsection: 'unit',
+            unit: 'component'
+        },
+
+        categoriesText: {
+            section: gettext('Sections'),
+            subsection: gettext('Subsections'),
+            unit: gettext('Units'),
+            component: gettext('Components')
+        },
+
+        events: {
+            'click .button-forward': 'renderChildren'
+        },
+
+        initialize: function(options) {
+            this.visitedAncestors = [];
+            this.template = HtmlUtils.template(MoveXBlockListViewTemplate);
+            this.ancestorInfo = options.ancestorInfo;
+            this.listenTo(Backbone, 'move:breadcrumbButtonPressed', this.handleBreadcrumbButtonPress);
+            this.renderXBlockInfo();
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(
+                this.$el,
+                this.template(
+                    {
+                        xblocks: this.childrenInfo.children,
+                        noChildText: this.getNoChildText(),
+                        categoryText: this.getCategoryText(),
+                        XBlocksCategory: this.childrenInfo.category,
+                        forwardButtonSRText: this.getForwardButtonSRText(),
+                        currentLocationIndex: this.getCurrentLocationIndex()
+                    }
+                )
+            );
+            Backbone.trigger('move:childrenRendered', this.breadcrumbInfo());
+            return this;
+        },
+
+        /**
+         * Forward button press handler. This will render all the childs of an XBlock.
+         *
+         * @param {Object} event
+         */
+        renderChildren: function(event) {
+            this.renderXBlockInfo(
+                'forward',
+                $(event.target).closest('.xblock-item').data('itemIndex')
+            );
+        },
+
+        /**
+         * Breadcrumb button press event handler. Render all the childs of an XBlock.
+         *
+         * @param {any} newParentIndex  Index of a parent XBlock
+         */
+        handleBreadcrumbButtonPress: function(newParentIndex) {
+            this.renderXBlockInfo('backward', newParentIndex);
+        },
+
+        /**
+         * Render XBlocks based on `forward` or `backward` navigation.
+         *
+         * @param {any} direction           `forward` or `backward`
```

Comment:
@andy-armstrong Depend upon forward/backward we decide whether we need to render parent or children. breadcrumb view has no information regarding forward/backward navigation, it just gives the index to list view to render the correct XBlock info.